src/errors.rs

```diff
@@ -0,0 +1,59 @@
+/// Errors that can occur during profiler creation.
+#[derive(thiserror::Error, Debug)]
+pub enum CreationError {
+    #[error(transparent)]
+    InvalidSettings(#[from] SettingsError),
+
+    #[cfg(feature = "tracy")]
+    #[error("Tracy client doesn't run yet.")]
+    TracyClientNotRunning,
+
+    #[cfg(feature = "tracy")]
+    #[error("Failed to create Tracy GPU context: {0}")]
+    TracyGpuContextCreationError(#[from] tracy_client::GpuContextCreationError),
+}
+
+impl PartialEq for CreationError {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            #[cfg(feature = "tracy")]
+            (Self::TracyGpuContextCreationError(left), Self::TracyGpuContextCreationError(right)) => match left {
+                tracy_client::GpuContextCreationError::TooManyContextsCreated => match right {
+                    tracy_client::GpuContextCreationError::TooManyContextsCreated => true,
+                },
+            },
+            _ => core::mem::discriminant(self) == core::mem::discriminant(other),
```

Comment:
Do we specifically want the different variants of SettingsError to be equivalent here?

Reply:
ah you're right this is actually not great. Need to do an exhaustive match. And/or submit a patch to tracy making their error `PartialEq`
Or maybe just drop it and use `matches!` where I needed it in the tests!

Reply:
Hmmm I was wondering what the PartiallEq was for. I think any of those 3 solutions would work.